### PR TITLE
association extensions

### PR DESCRIPTION
**Pull request by singhal-ekansh**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
+  has_many :products, through: :line_items
+  has_many :enabled_products, -> { where("enabled is null") }, source: :product, through: :line_items
```
Comment (harsh-ag) [inline app/models/cart.rb]: use rails syntax in where query

File: db/migrate/20230411171447_rename_column_name_in_carts.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class RenameColumnNameInCarts < ActiveRecord::Migration[7.0]
+  def change
+    rename_column :carts, :line_item_count, :line_items_count
```
Comment (harsh-ag) [inline db/migrate/20230411171447_rename_column_name_in_carts.rb:3]: where's the migration to add column `line_item_count`

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -61,6 +61,16 @@ def destroy
     redirect_to users_url, notice: exception.message
   end
 
+  def show_orders
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb]: rename to orders

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -61,6 +61,16 @@ def destroy
...
+    @user = User.find_by(id: session[:user_id])
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb]: fetch orders here and use before_action to set user

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -61,6 +61,16 @@ def destroy
     redirect_to users_url, notice: exception.message
   end
 
+  def show_orders
+    @user = User.find_by(id: session[:user_id])    
+  end
+
+  def show_line_items
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb]: rename to line_items

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -61,6 +61,16 @@ def destroy
...
+    @page_number = params[:page] ? params[:page].to_i : 1 
+    @user = User.find_by(id: session[:user_id])
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb]: use before_action

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -61,6 +61,16 @@ def destroy
...
   end
 
+  def show_orders
+    @user = User.find_by(id: session[:user_id])    
+  end
+
+  def show_line_items
+    @page_number = params[:page] ? params[:page].to_i : 1 
+    @user = User.find_by(id: session[:user_id])   
+    @line_items = @user.line_items.limit(5).offset(5 * (@page_number-1))
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb]: use constant to define the line items per page

**Review by harsh-ag — CHANGES_REQUESTED** [review]

File: db/migrate/20230411171447_rename_column_name_in_carts.rb (hunk 2 above)
Comment (singhal-ekansh) [reply db/migrate/20230411171447_rename_column_name_in_carts.rb:3]: sir line_items_count was to be added in migration exercise but that time by mistake i had added line_item_count

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show edit update destroy ]
+  before_action :set_logged_user, only: [:orders, :line_items]
```
Comment (harsh-ag) [inline app/controllers/users_controller.rb:3]: @singhal-ekansh why can't we use the existing `set_user` method.

**Review by harsh-ag — APPROVED** [review]

File: app/controllers/users_controller.rb (hunk 8 above)
Comment (singhal-ekansh) [reply app/controllers/users_controller.rb:3]: will change after reading routes

**Pull request closed without merging**